lib/ansible/modules/cloud/packet/packet_device.py

```diff
@@ -445,6 +445,11 @@ def create_single_device(module, packet_conn, hostname):
     locked = module.params.get('locked')
     ipxe_script_url = module.params.get('ipxe_script_url')
     always_pxe = module.params.get('always_pxe')
+    if operating_system != 'custom_ipxe':
+        for param in ('ipxe_script_url', 'always_pxe'):
+            if module.params.get(param):
+                raise Exception('%s paramater is not valid for non custom_ipxe operating_system.' % param)
```

Comment:
I would recommend to use `module.fail_json` instead of raising an exception here.

Reply:
will do, I'll update the other cases of Exception in the files here too

Reply:
@Akasurde hmm idk, any reason why I should do that here instead of following the rest of the file and letting the try/except catch it there?

Reply:
my bad.